pkg/reconciler/autoscaling/kpa/scaler.go

```diff
@@ -307,6 +308,17 @@ func (ks *scaler) scale(ctx context.Context, pa *pav1alpha1.PodAutoscaler, sks *
 	}
 
 	min, max := pa.ScaleBounds()
+	initialScale := kparesources.GetInitialScale(config.FromContext(ctx).Autoscaler, pa)
+	sw := config.FromContext(ctx).Autoscaler.StableWindow
+	hasBeenActiveFor := pa.Status.HasBeenActiveFor(time.Now())
+	if initialScale > 1 && hasBeenActiveFor < sw {
```

Comment:
I don't quite get why this is relative to the stable window?

Shouldn't we just apply `min = int32(intMax(initialScale, min))` if `!pa.Status.HasBeenActive()`?

Reply:
If we get rid of `hasBeenActiveFor < sw`, we still have the problem of scaling down immediately after hitting initialScale, because `scaler.scale(...)` is triggered after receiving a new desiredScale. If initialScale = 10, as soon as we scale to 10, the desiredScale will be 5 then we scale down immediately here.

Reply:
Not if we only apply it if we have not been ready? :thinking:

Reply:
Hmm I'm not sure I'm following. We do currently mark HasBeenReady as soon as the initialScale is reached, so in `scale(...)` we will apply the new desiredScale if we get rid of the stable window check right?

Reply:
@markusthoemmes  I think `!pa.Status.HasBeenActive()` is covered with `hasBeenActiveFor` returning 0 when PA has never been active at all.

But this makes sense if we agree that `we still have the problem of scaling down immediately after hitting initialScale` is a problem in itself.

Reply:
This is the main remaining issue. I am not sure that we need to keep all the pods for the stable window if there's no traffic (since if there's is adequate traffic it won't scale down?)
A real-life test is appropriate here.

Reply:
When I tested this locally it did scale down immediately from initialScale. If we were to remove the stable window part, I can't think of a way to verify this feature works correctly in an e2e test.

Reply:
Would starting in panic mode not avoid this issue?

Reply:
What I mean is if you send traffic to the new revision?
@julz we decided to avoid all changes to autoscaler core altogether for this change.

Reply:
@julz yes starting in panic mode would solve this issue, but we removed the panic mode part and are keeping code changes out of autoscaler.

Reply:
Im guessing the problem with sending traffic is it's hard to distinguish whether you hit the required scale via this feature, or just because of the traffic. I guess you could test that there's no window after going Ready where it's less than initialScale, but it's inevitably a little racy. I guess if we set maxScaleUpRate low it might be reliable enough?

(the main reason I ask about panic mode is it sort of seems like we're implementing start-in-panic-mode by the back door here anyway, but if we can do it with just sending traffic then cool)

Reply:
The way I look at this, as a special case for `minScale` and hence it feels very natural to be right here, next to it.
The broader question is: _is it a problem if we immediately scale down_? Which I am not sure we know yet :)